engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -43,6 +42,7 @@ object SlickDataAccess {
   private lazy val rootDatabaseConfig = rootConfig.getConfig("database")
   private lazy val databaseConfigName = rootDatabaseConfig.getStringOption("config")
   lazy val defaultDatabaseConfig = databaseConfigName.map(getDatabaseConfig).getOrElse(rootDatabaseConfig)
+  val actorSystem = ActorSystem("slick-retry-scheduler")
```

Comment:
Ask for the actor system in your method parameters. I could even sign off on a patch that implements two different `runTransaction` methods for this patch. 1) a new one that retries by scheduling a message to come back over the actor system, 2) the original that just tries once and fails. The former would only be used by the upsert methods for now, the latter by everything else.

For the love of all that's good, please don't use `CromwellServer.actorSystem`. That singleton needs go away asap or undergo refactoring as it's not even supposed to be around during the current 'cromwell run my.wdl'.